From 1ffd1039b3e38822106cac2d7326a8e3ddaa18e5 Mon Sep 17 00:00:00 2001
From: Boon Khai Ng <boon.khai.ng@intel.com>
Date: Thu, 26 Aug 2021 11:36:26 +0800
Subject: [PATCH 36/42] HSD #14014974605: crypto: intel_fcs: Update Certificate
 Mask for FPGA Attestation

commit 60722e6cd43073e9ee7141011132e36157773f5e from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

Update the certificate mask to 0x001f to cover the Alias Certificate
in Agilex family.

Signed-off-by: Boon Khai Ng <boon.khai.ng@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/crypto/intel_fcs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/intel_fcs.c b/drivers/crypto/intel_fcs.c
index 3708e97cb675..d9bb528d2c82 100644
--- a/drivers/crypto/intel_fcs.c
+++ b/drivers/crypto/intel_fcs.c
@@ -983,7 +983,7 @@ static long fcs_ioctl(struct file *file, unsigned int cmd,
 		msg->payload_length = 0;
 		msg->payload_output = d_buf;
 		msg->payload_length_output = CERTIFICATE_RSP_MAX_SZ;
-		msg->arg[0] = data->com_paras.certificate.c_request & 0x000f;
+		msg->arg[0] = data->com_paras.certificate.c_request & 0x001f;
 		priv->client.receive_cb = fcs_attestation_callback;
 
 		ret = fcs_request_service(priv, (void *)msg,
@@ -1022,7 +1022,7 @@ static long fcs_ioctl(struct file *file, unsigned int cmd,
 		}
 
 		msg->command = COMMAND_FCS_ATTESTATION_CERTIFICATE_RELOAD;
-		msg->arg[0] = data->com_paras.c_reload.c_request & 0x000f;
+		msg->arg[0] = data->com_paras.c_reload.c_request & 0x001f;
 		priv->client.receive_cb = fcs_vab_callback;
 		ret = fcs_request_service(priv, (void *)msg,
 					  10 * FCS_REQUEST_TIMEOUT);
-- 
2.31.1

